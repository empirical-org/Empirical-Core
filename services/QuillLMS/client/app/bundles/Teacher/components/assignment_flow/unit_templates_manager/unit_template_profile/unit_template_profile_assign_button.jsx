import React from 'react';
import { requestPost, } from '../../../../../../../app/modules/request'
import AnalyticsWrapper from '../../../shared/analytics_wrapper';
import { UNIT_TEMPLATE_NAME, } from '../../assignmentFlowConstants'

export default class UnitTemplateProfileAssignButton extends React.Component {
  analytics = () => {
    return new AnalyticsWrapper();
  };

  saveAnonymousAssignmentValues = (redirect) => {
    const { data, } = this.props
    window.localStorage.setItem(UNIT_TEMPLATE_NAME, data.name)
    requestPost(
      '/session/set_post_auth_redirect',
      { post_auth_redirect: window.location.href },
      () => window.location.href = redirect
    );
  }

  handleClickLogIn = () => this.saveAnonymousAssignmentValues('/session/new')

  handleClickSignUp = () => this.saveAnonymousAssignmentValues('/account/new')

  propsSpecificComponent = () => {
    if (!this.props.data.non_authenticated) { return }

    return (
      <div className="login-or-signup-container light-gray-bordered-box">
        <strong>Log in or sign-up to assign</strong>
        <div className="login-or-signup-buttons">
          <button className="quill-button medium secondary outlined" onClick={this.handleClickLogIn}>Log in</button>
          <button className="quill-button medium primary contained" onClick={this.handleClickSignUp}>Sign up</button>
        </div>
      </div>
    )
  };

  render() {
    return (
      <div>
        {this.propsSpecificComponent()}
        <p className="time"><i className="far fa-clock" />Estimated Time: {this.props.data.time} mins</p>
      </div>
    );
  }
}
