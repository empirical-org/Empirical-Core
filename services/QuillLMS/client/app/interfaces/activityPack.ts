import React from "react";

export interface DropdownObject {
  value: string,
  label: string
}

export interface Classroom {
  classroom: {
    clever_id: number,
    code: string,
    created_at: string,
    google_classroom_id: number,
    grade: string,
    grade_level: number,
    id: number,
    name: string,
    synced_name: string,
    teacher_id: number,
    updated_at: string,
    visible: boolean
  }
  students: {
    account_type: string,
    active: boolean,
    classcode: string,
    clever_id: number,
    created_at: string,
    email: string
    flags: string[],
    google_id: number,
    id: number
    ip_address: {
      family: number,
      addr: number,
      mask_addr: number
    }
    last_active: string,
    last_sign_in: string,
    name: string,
    password_digest: string,
    post_google_classroom_assignments: boolean,
    role: string,
    send_newsletter: boolean,
    signed_up_with_google: boolean,
    stripe_customer_id: number,
    time_zone: string,
    title: string,
    token: string,
    updated_at: string,
    username: string,
  }[]
 }

 export interface ClassroomUnit {
  assign_on_join: boolean,
  assigned_student_ids: number[],
  classroom_id: number,
  created_at: string,
  id: number,
  unit_id: number,
  updated_at: string,
  visible: boolean
}

export interface Activity {
  activity_category: { id: number, name: string },
  activity_category_id: number,
  activity_category_name: string,
  activity_classification: {
    alias: string,
    description: string,
    key: string,
    id: number
  }
  anonymous_path: string,
  content_partners: any[],
  description: string,
  flags: string,
  id: number
  name: string,
  readability_grade_level: string,
  standard_level: { id: number, name: string }
  standard_level_name: string,
  standard_name: string,
  topics: {
    id: number,
    level: number,
    name: string,
    parent_id: number
  }[],
  uid: string
}

// activity object from DataTable
export interface ActivityElement extends Activity {
  // activityClassificationId: number,
  activityId?: number,
  // classroomId: number,
  // createdAt: string,
  // created_at: number,
  cuId?: number,
  // dueDate: string
  // dueDatePicker: JSX.Element,
  // id: number,
  // name: string,
  // ownedByCurrentUser: true
  // ownerName: string,
  // removable: true
  // shareActivity: JSX.Element,
  // toolAndNameSection: JSX.Element,
  // uaId: number
}

export interface ActivityPack {
  activities: Activity[],
  activityCount: number,
  name: string
}
