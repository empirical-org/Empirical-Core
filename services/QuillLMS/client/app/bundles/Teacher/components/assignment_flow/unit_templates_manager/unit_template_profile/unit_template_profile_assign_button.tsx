import * as React from 'react';
import { Activity } from '../../../../../../interfaces/activityPack';

import { EVIDENCE } from '../../../../../Shared';

interface DataInterface {
  activities: Activity[],
  activity_info: string,
  created_at: number,
  diagnostics_recommended_by: Activity[]
  flag: string,
  grades: string[],
  id: number
  name: string,
  non_authenticated: boolean,
  number_of_standards: number,
  order_number: number,
  readability: string,
  time: number,
  type: {
    name: string,
    primary_color: string
  }
  unit_template_category: {
    primary_color: string,
    secondary_color: string,
    name: string,
    id: number
  }
}
interface UnitTemplateProfileDisclaimerPropsInterface {
  data: DataInterface
}
export const UnitTemplateProfileDisclaimer = ({ data }: UnitTemplateProfileDisclaimerPropsInterface) => {

  function isEvidenceActivityPack() {
    if(!data || !data.activities) { return false }

    const { activities } = data;
    return activities.some(activity => {
      const { classification } = activity;
      const { key } = classification;
      return key === EVIDENCE;
    });
  }

  const showWarning = isEvidenceActivityPack();

  return(
    <section className="evidence-warning-container">
      {showWarning && <section className="evidence-warning">
        <p className="header">Activity Difficulty: Designed for 8th-12th Grade</p>
        <p className="text">Quill Reading for Evidence activities are designed for 8th-12th grade students or students reading at a Lexile level between 950-1250.</p>
      </section>}
      <p className="time"><i className="far fa-clock" />Estimated Time: {data.time} mins</p>
    </section>
  );
}

export default UnitTemplateProfileDisclaimer
