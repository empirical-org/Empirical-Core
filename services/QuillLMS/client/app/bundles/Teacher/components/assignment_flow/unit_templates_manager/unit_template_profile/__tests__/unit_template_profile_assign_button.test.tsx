import * as React from 'react';
import { mount } from 'enzyme';

import { UnitTemplateProfileAssignButton } from '../unit_template_profile_assign_button';
import { activityPack, evidenceActivity } from '../../../../../../../test_data/activityPack';

const mockProps = {
  data: activityPack
}
describe('UnitTemplateProfileDescription component', () => {
  let component = mount(
    <UnitTemplateProfileAssignButton {...mockProps} />
  );
  it('should match snapshot', ()=> {
    expect(component).toMatchSnapshot();
  })
  it('should show login and signup buttons if non_authenticated is true', ()=> {
    expect(component.find('.login-or-signup-buttons').length).toEqual(0);
    mockProps.data.non_authenticated = true;
    component = mount(<UnitTemplateProfileAssignButton {...mockProps} />);
    expect(component.find('.login-or-signup-buttons').length).toEqual(1);
  })
  it('should show Evidence warning if at least one Evidence activity is present', ()=> {
    expect(component.find('.evidence-warning').length).toEqual(0);
    mockProps.data.activities.push(evidenceActivity);
    component = mount(<UnitTemplateProfileAssignButton {...mockProps} />);
    expect(component.find('.evidence-warning').length).toEqual(1);
  })
});
