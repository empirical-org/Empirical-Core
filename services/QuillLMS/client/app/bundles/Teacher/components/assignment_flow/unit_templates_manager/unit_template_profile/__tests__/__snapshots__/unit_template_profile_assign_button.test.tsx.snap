// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnitTemplateProfileDescription component should match snapshot 1`] = `
<UnitTemplateProfileDisclaimer
  data={
    Object {
      "activities": Array [
        Object {
          "classification": Object {
            "id": 1,
            "key": "sentence",
            "name": "Quill Grammar",
          },
          "description": "This is a test Grammar activity.",
          "id": 1,
          "level_zero_topic_name": null,
          "name": "Their, They're, There",
          "readability": "4th-5th",
          "standard": Object {
            "id": 1,
            "name": "1.1 Test Grammar Standard",
            "standard_category": Object {
              "id": 1,
              "name": "Test Grammar Standard",
            },
          },
          "standard_level_name": "CCSS: Grade 4",
        },
      ],
      "activity_info": "",
      "created_at": 123123123,
      "diagnostics_recommended_by": null,
      "flag": "",
      "grades": Array [],
      "id": 3,
      "name": "Test Activity Pack",
      "non_authenticated": false,
      "number_of_standards": 0,
      "order_number": 0,
      "readability": "",
      "time": 60,
      "type": Object {
        "name": "",
        "primary_color": "",
      },
      "unit_template_category": Object {
        "id": 4,
        "name": "",
        "primary_color": "",
        "secondary_color": "",
      },
    }
  }
>
  <section
    className="evidence-warning-container"
  >
    <p
      className="time"
    >
      <i
        className="far fa-clock"
      />
      Estimated Time: 
      60
       mins
    </p>
  </section>
</UnitTemplateProfileDisclaimer>
`;
